docs/MNIST.md

```diff
@@ -0,0 +1,371 @@
+
+
+```python
+from tensorflow.examples.tutorials.mnist import input_data
+```
+
+
+```python
+mnist = input_data.read_data_sets("MNIST_data/", one_hot=True)
+```
+
+    Extracting MNIST_data/train-images-idx3-ubyte.gz
+    Extracting MNIST_data/train-labels-idx1-ubyte.gz
+    Extracting MNIST_data/t10k-images-idx3-ubyte.gz
+    Extracting MNIST_data/t10k-labels-idx1-ubyte.gz
+
+
+
+```python
+import deepchem as dc
+import tensorflow as tf
+from deepchem.models.tensorgraph.layers import Layer, Input, Reshape, Flatten
+from deepchem.models.tensorgraph.layers import Dense, SoftMaxCrossEntropy, ReduceMean, SoftMax
+```
+
+
+```python
+train = dc.data.NumpyDataset(mnist.train.images, mnist.train.labels)
+valid = dc.data.NumpyDataset(mnist.validation.images, mnist.validation.labels)
+```
+
+
+```python
+tg = dc.models.TensorGraph(tensorboard=True, model_dir='/tmp/mnist')
+feature = Input(shape=(None, 784))
+tg.add_layer(feature)
+tg.add_feature(feature)
+
+# Images are square 28x28 (batch, height, width, channel)
+make_image = Reshape(shape=(-1, 28, 28, 1))
+tg.add_layer(make_image, parents=[feature])
+
+class Conv2d(Layer):
```

Comment:
We should probably move `Conv2d` directly into DeepChem, especially since we want to eventually support more visual/microscopy datasets moving forward.